src/components/subscriptions/buttons/DownloadCsvButton.jsx

```diff
@@ -0,0 +1,57 @@
+import React, { useContext, useState } from 'react';
+import { Icon, StatefulButton } from '@edx/paragon';
+import { logError } from '@edx/frontend-platform/logging';
+import { saveAs } from 'file-saver';
+import { SubscriptionDetailContext } from '../SubscriptionDetailContextProvider';
+import LicenseManagerApiService from '../data/service';
+
+const DownloadCsvButton = () => {
+  const { subscription } = useContext(SubscriptionDetailContext);
+  const [buttonState, setButtonState] = useState('default');
+
+  const getCsvFileName = () => {
+    const titleNoWhitespace = subscription.title.replace(/\s+/g, '');
+    const currentDate = new Date();
+    const year = currentDate.getUTCFullYear();
+    const month = currentDate.getUTCMonth() + 1;
+    const day = currentDate.getUTCDate();
+    return `${titleNoWhitespace}-${year}-${month}-${day}.csv`;
+  };
+
+  return (
+    <StatefulButton
+      state={buttonState}
+      variant="outline-primary"
+      labels={{
+        default: 'Download CSV',
+        pending: 'Downloading',
+        complete: 'Downloaded',
+      }}
+      icons={{
+        default: <Icon className="fa fa-download" />,
+        pending: <Icon className="fa fa-spinner fa-spin" />,
+        complete: <Icon className="fa fa-check" />,
```

Comment:
nit: It's preferable to use SVG icons vs. icons controlled via classes (i.e., a font-based icon) as they are more scalable and less prone to charset issues (utf-8). Unfortunately, the Paragon `Icon` component doesn't support SVG icons from font-awesome as it is intended to be used with the icons hosted within Paragon itself (`@edx/paragon/icons`).

Example:

```jsx
import { faDownload, faSpinner } from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

return (
  <StatefulButton
    icons={{
     default: <Icon src={faDownload} />,
     pending: <Icon src={faSpinner} className="fa-spin" />,
     ...
    }}
  />
);
```